zilliqa/src/consensus.rs

```diff
@@ -729,6 +644,10 @@ impl Consensus {
             )
         })?;
 
+        // Tell the transaction pool that the sender's nonce has been incremented.
```

Comment:
Good point, does
```rust
        if result.success {
            self.transaction_pool.update_nonce(&txn);
        }
```
make sense?

Reply:
I don't think that would be correct:
```
                // Note that success can be false, the tx won't apply changes, but the nonce increases
                // and we get the return value (which will indicate the error)
                
 ```
                
Basically, if your TX reverts/runs out of gas or is otherwise bad, you still get charged for the gas used, so the account nonce should increase.

Reply:
Then surely the previous version is correct? If I'm reading correctly `fn apply_transaction` will increment the nonce IFF it returns `Ok(_)`. And thus we should be fine to unconditionally call `self.transaction_pool.update_nonce(&txn);` here after `apply_transaction(...)?` has returned?